File: tests/fixtures/sample-python-project-il/code/app/services/customers_service.py
"""Customer creation."""

from datetime import datetime
from uuid import uuid4

from app.orm import Customer


def create_customer(email: str, name: str):
    # Spec says email is lowercased on write. Storing raw input means two
    # customers with `Foo@Example.com` and `foo@example.com` are not deduped
    # and lowercase-email lookups miss the row.
    # IL-DRIFT: Entity:Customer / field.email.normalize
    customer = Customer(
        id=str(uuid4()),
        email=email,
        name=name,
        loyalty_tier="standard",
        status="active",
        created_at=datetime.utcnow().isoformat(),
    )
    return customer

File: tests/fixtures/sample-csharp-project-il/code/Services/CustomersService.cs
using SampleApi.Data.Entities;

namespace SampleApi.Services;

public class CustomersService
{
    public Customer CreateCustomer(string email, string name)
    {
        // Spec says email is lowercased on write. Storing raw input means two
        // customers with `Foo@Example.com` and `foo@example.com` are not deduped
        // and lowercase-email lookups miss the row.
        // IL-DRIFT: Entity:Customer / field.email.normalize
        return new Customer
        {
            Id = Guid.NewGuid(),
            Email = email,
            Name = name,
            LoyaltyTier = "standard",
            Status = "active",
            CreatedAt = DateTime.UtcNow,
        };
    }
}
